src/main/java/Domain/Controllers/TeamController.java

```diff
@@ -33,18 +36,45 @@ public static boolean addTeamOwner(String username, Team teamToOwn, TeamOwner ow
         {
 
             teamOwner = (TeamOwner) newTeamOwnerRole;
+
             if(teamOwners.contains(teamOwner))
             {
-                throw new Exception("This User is already a team owner");
+                throw new Exception("This User is already a team owner of this team");
+            }
+
+            if(isAlreadyOwnedAnotherTeamInLeague(teamToOwn,teamOwner)){
+                throw new Exception("This User is already a team owner of a different team in same league");
             }
+
         }
-        teamOwner.addTeamToOwn(teamToOwn);
-        teamToOwn.addTeamOwner(owner,teamOwner);
 
+        teamOwner.addTeamToOwn(teamToOwn);
+        teamToOwn.addTeamOwner(teamOwner);
 
 
         return true;
     }
 
+    /**
+     *Check if the new team owner already owned a team in the same league the team he should be owned exist
+     * @param teamToOwn the team he should be owned to
+     * @param ownerToCheck the new team owner
+     * @return true if the new team owner already owned a team in the same league
+     */
+    private static boolean isAlreadyOwnedAnotherTeamInLeague(Team teamToOwn,TeamOwner ownerToCheck){
+
+        Season currentSeason = teamToOwn.getCurrentSeason();
```

Comment:
The name of the function is League but you checked season is it the same?

Reply:
A season belongs to a league, so I think this is fine. We don't need to check all of the league's seasons.

Reply:
yes you right, I need to change the name of the function and the JavaDoc

Reply:
I see you didnt change it...are we leaving it League or not?

Reply:
Fixed